runtime/runtime_test.go

```diff
@@ -323,54 +324,54 @@ func TestEVMFunction(t *testing.T) {
 			},
 		},
 		{
-			name:       "pre ETH_SH basefee",
-			code:       "6080604052348015600e575f80fd5b50600436106026575f3560e01c80635cf2496914602a575b5f80fd5b60306044565b604051603b91906061565b60405180910390f35b5f48905090565b5f819050919050565b605b81604b565b82525050565b5f60208201905060725f8301846054565b9291505056fea2646970667358221220cbae2729fc31c76b30a14b2a0f61fdad920cb3c0ef1f7f5166e48d836d91cc6a64736f6c63430008180033",
-			abi:        `[{"inputs":[],"name":"basefee","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"}]`,
-			methodName: "basefee",
+			name:       "ETH_SH PUSH0 gas cost",
+			code:       "",
+			abi:        "",
+			methodName: "",
 			testFunc: func(ctx *context, t *testing.T) {
-				// pragma solidity >=0.7.0 <0.9.0;
-				// contract TestBaseFee {
-				//     function basefee() public view returns (uint256) {
-				//         return block.basefee;
-				//     }
-				// }
-				methodData, err := ctx.method.EncodeInput()
-				if err != nil {
-					t.Fatal(err)
-				}
+				// 0x5f is PUSH0 opCode
+				codeData := []byte{0x5f, 0x00}
+				exec, _ := runtime.New(ctx.chain, ctx.state, &xenv.BlockContext{}, thor.ForkConfig{}).
+					PrepareClause(tx.NewClause(nil).WithData(codeData), 0, math.MaxUint64, &xenv.TransactionContext{})
+				out, _, err := exec()
 
+				assert.Nil(t, err)
+				assert.Nil(t, out.VMErr)
+				assert.Equal(t, uint64(2), math.MaxUint64-out.LeftOverGas)
+			},
+		},
+		{
+			name:       "pre ETH_SH basefee",
+			code:       "4800",
+			abi:        "",
+			methodName: "",
+			testFunc: func(ctx *context, t *testing.T) {
+				// 0x48 is BASEFEE opCode
+				codeData := []byte{0x48, 0x00}
 				exec, _ := runtime.New(ctx.chain, ctx.state, &xenv.BlockContext{}, thor.NoFork).
-					PrepareClause(tx.NewClause(&target).WithData(methodData), 0, math.MaxUint64, &xenv.TransactionContext{})
+					PrepareClause(tx.NewClause(&target).WithData(codeData), 0, math.MaxUint64, &xenv.TransactionContext{})
```

Comment:
I think for this test `codeDate` is not required, right?

Reply:
correct, will remove